File: src/codility/lesson1/iterations.py
import re

def solution( N ):

    binary = "{0:b}".format( N )
    # print "binary : {0}".format( binary )

    regex = re.compile( '10+1' )

    gaps = []
    while regex.search( binary ):
        match = regex.search( binary )
        str = match.group()
        # print str
        gaps.append( len( str ) - 2 )
        binary = binary.replace( str, "11", 1 )

    if len( gaps ) == 0:
        gaps.append( 0 )
    
    gaps.sort( reverse = True )

    return gaps[ 0 ]

result = solution( 2147083640 )
print( "### The largest gap is {0}".format( result ) )

File: src/codility/lesson1/iterations.js
let binary = "10010101000100";

let regex = /10*1/;
let searched = regex.exec( binary );
console.log( searched );

let gaps = [];
while( ( searched = regex.exec( binary ) ) != null ) {
    let gap = searched.toString().length - 2;
    console.log( `### gap : ${gap}` );
    gaps.push( gap );
    binary = binary.replace( searched, `1` );
}

gaps.sort( ( a, b ) => b - a );
if ( gaps.length == 0 ) {
    gaps.push( 0 );
}
console.log( `### The largest gap is ${gaps[ 0 ]}` );

// console.log( binary );